### PR TITLE
Better controller for plex.

### PR DESCRIPTION
**Pull request by Hellowlol**
Add a enchanted Plex controller with additional methods.

There is also a plexapi controller that have some additional features, like change subtitle, audio etc. It isnt the cleanest method to do this but i didnt see any other way to add support for this.

Note:
This pr will not require that plexapi is a requirement pychromecast. All methods in PlexController can be used without plexapi, but using it make the usage alot easier and there is less stuff to set manually.

### CODE REVIEW

**Comment by Hellowlol** [comment]
If someone please would tell me how i can make travis happy? please let me know so i can rebase this thing.

File: pychromecast/controllers/plex.py [hunk 1]
```diff
@@ -1,40 +1,538 @@
...
+    @property
+    def status(self):
+        """Get the chromecast playing status.
+
+        Returns:
+            pychromecast.controllers.media.MediaStatus: Slightly modified status with patched
+                                                        method for episode_title.
+        """  # noqa
+
+        @property
```
Comment (balloob) [inline pychromecast/controllers/plex.py]: I don't think that this should be defined inside `status` ?

File: pychromecast/controllers/plex.py [hunk 2]
```diff
@@ -1,40 +1,538 @@
...
+
+            self._send_cmd(
+                msg,
+                namespace="urn:x-cast:com.google.cast.media",
+                inc_session_id=True,
+                inc=False,
+            )
+        else:
+            self.logger.debug(
+                "Cant reset teh stream as _last_play_msg "
```
Comment (balloob) [inline pychromecast/controllers/plex.py]:
```suggestion
                "Cant reset the stream as _last_play_msg "
```

File: pychromecast/controllers/plex.py [hunk 3]
```diff
@@ -1,40 +1,538 @@
...
+
+    def disconnect(self, timeout=None, blocking=True):
+        """Disconnect the controller."""
+        self._socket_client.disconnect()
+        if blocking:
+            self.join(timeout=timeout)
+
+
+# pylint: disable=too-many-public-methods
+class PlexApiController(PlexController):
```
Comment (balloob) [inline pychromecast/controllers/plex.py]: Why does this class exist? `self.pms` is not being used.

Comment (Hellowlol) [reply pychromecast/controllers/plex.py]: Its being used inside _get_current_media, I have separated this a in its own class as I haven't found another way to to do the public methods in this class without plexapi.

File: pychromecast/controllers/plex.py (hunk 1 above)
Comment (Hellowlol) [reply pychromecast/controllers/plex.py]: Do you want me to add this to MediaStatus or just define in outside the property?

**Comment by Hellowlol** [comment]
Is there anything else i need to do to get this merged?

**Comment by balloob** [comment]
Looks good! Thanks

**Comment by Nojahhh** [comment]
Thanks for a great update on the PlexController @Hellowlol.

Tried this out and its working great with plexapi.

**Comment by Hellowlol** [comment]
@Nojahhh Thanks! It's nice to know others could benefit. I needed chromecast support for my own project :)

**Pull request merged**